src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12367,7 +12367,7 @@ private InitializerExpressionSyntax ParseObjectOrCollectionInitializer()
                 SkipBadInitializerListTokens,
                 allowTrailingSeparator: true,
                 requireOneElement: false,
-                allowSemicolonAsSeparator: false);
+                allowSemicolonAsSeparator: true);
```

Comment:
why was this cahnge alone not sufficient to making things better?

Reply:
See the other thread, but this turned out to be more nuanced than that.

Reply:
is it simply that isPossibleElement is improperly consuming too much?   should we change the isPossibleElement for this initializer case to not consume in those other cases? Basically, narrow the change to just what ParseObjectOrCollectionInitializer does, instead of the core separated syntax list parsing code.  thankS!

Reply:
Sorry for the delay, I had to recheck since it's been a bit I opened this pr, but I am very confused as to why IsPossibleElement is involved here.

The parsing stop occurs as part of the skipBadTokens because when it encounters a semicolon while it's parsing a declaration, it think it's terminal because the declaration is supposed to end with it. This is ofc not what happens here, but it affects the syntax tree badly because the parser now thinks that the declaration actually ended before the closing brace tokens. This cascades into the error propagation I talked about in the issue thread.

So I am not sure why IsPossibleElement is involved here: from what I can tell, all that one really does is check if the current token could be an element. I could make it look ahead by one token, but the problem is that will prevent parsing the last element if it returns false when it sees the semicolon. The issue occurs specifically after the last element was parsed and it expects the separator.

Reply:
@CyrusNajmabadi Hey, my question is still valid above here so just sending a reminder on it.

Reply:
i'm basically referring to IsInitializerMember.  I'm not sure why we're not just fixing htat up to not consume too much.

Reply:
@CyrusNajmabadi what counsumption are you reffering to?

I check the method, all it does is check the current tokens and peeks ahead, but it never consumes any tokens. I am confused on what exactly you mean by the method consuming too much when it doesn't consume any.